core/src/main/java/tc/oc/pgm/loot/LootableMatchModule.java

```diff
@@ -0,0 +1,337 @@
+      // of Inventories that they are allowed to fill, based on the filter of each Filler.
+      // Note how duplicate inventories are skipped.
+      final SetMultimap<FillerDefinition, Inventory> fillerInventories = HashMultimap.create();
+      inventories()
+          .distinct()
+          .forEach(
+              inventory -> {
+                final Predicate<Filter> passes = passesFilter(inventory.getHolder());
+                for (FillerDefinition filler : coolFillers) {
+                  if (passes.test(filler.getFilter())) {
+                    fillerInventories.put(filler, inventory);
+                  }
+                }
+              });
+
+      // Do all clearing before we start filling anything.
+      // Undefined behaviour here when multiple fillers with dynamic trigger are
+      // executed at the same time
+      fillerInventories
+          .asMap()
+          .forEach(
+              (filler, inventories) -> {
+                if (filler.cleanBeforeRefill()) {
+                  inventories().forEach(Inventory::clear);
+                }
+              });
+
+      final Random random = new Random();
+
+      fillerInventories
+          .asMap()
+          .forEach(
+              (filler, inventories) -> {
+                // For each Filler, build a mutable list of slots that it can fill.
+                // (27 is the standard size for single chests)
+                final List<InventorySlot> slots = new ArrayList<>(inventories.size() * 27);
+                inventories.forEach(
+                    inv -> {
+                      for (int index = 0; index < inv.getSize(); index++) {
+                        if (inv.getItem(index) == null) {
+                          slots.add(new InventorySlot(Slot.forInventoryIndex(inv, index), inv));
+                        }
+                      }
+                    });
+
+                if (!slots.isEmpty()) {
+                  // Generate the loot for this Filler
+                  boolean repeat;
+                  do {
+                    filler
+                        .getLoot()
+                        .elements(opener)
+                        .forEachOrdered(
+                            item -> {
+                              if (!slots.isEmpty()) {
+                                // For each item, remove a random slot from those remaining,
+                                // apply item mods, and put it in the slot.
+                                if (immm != null) immm.applyRules(item);
+                                slots.remove(random.nextInt(slots.size())).putItem(item);
+                              }
+                            });
+                    int newValue =
+                        triggerHistory.row(this).compute(filler, (f, i) -> i == null ? 1 : ++i);
+                    repeat = filler.usesRefillTrigger() && newValue < triggeredFillers.get(filler);
+                  } while (repeat);
+                }
```

Comment:
I don't get why this loop is needed

1.12 pgm makes this logic work so that if there's a refill trigger, when it triggers it removes older filled at for that same filler, and doesn't seem to do anything else with it, what am i missing?

Reply:
Yes, but as I mentioned in the PR and in the javadocs on this class I changed the behavior for this in the new proto to apply multiple times(multiple rises). Maybe it should be an attribute instead?

Reply:
Was this behavior not at all present in 1.12 pgm? If so, can you elaborate as to why we want this change? why is it proto-dependant instead of just an xml flag? does it add extra value that original impl did not have? 

If this behavior is not from 1.12 PGM, next proto should be 1.4.4, as 1.4.3 was used in 1.12 pgm

Reply:
Not present in project ares pgm.

The more I think about this the more I think the multiple rise/multiple fill thing should be a xml flag. Why I think it should be a thing is because in my opinion it makes sense for it to work especially in scenarios where the trigger is a non-time trigger (e.g completing some objective multiple times) you still want to reward your players multiple times instead of "the amount of times they have opened the lootable chest in between each objective completion"

I still believe proto change makes sense for the explicit fill at match start makes sense. If lootables are used in any other scenarios than maps like SG or Skywars where _all_ chests are used for items throughout the match there is a big probability you don't want it to fill it at match start, but since it is legacy behavior if it is an attribute it has to default to true, which I think does not makes sense